proposals/extensions.md

```diff
@@ -38,7 +38,6 @@ extension_member_declaration // add
     | property_declaration
     | indexer_declaration
     | operator_declaration
-    | type_declaration
```

Comment:
:memo: removed extension types for now. We have an encoding solution available so no need to front-load extra design that doesn't seem to have unique challenges.